src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -165,3 +193,21 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_ipc_ep_put_zcopy,
                                 uct_iov_total_length(iov, iovcnt));
     return status;
 }
+
+ucs_status_t uct_cuda_ipc_ep_check(const uct_ep_h tl_ep, unsigned flags,
+                                   uct_completion_t *comp)
+{
+    uct_cuda_ipc_ep_t *ep = ucs_derived_of(tl_ep, uct_cuda_ipc_ep_t);
+    CUdevice this_device;
+
+    uct_cuda_ipc_ep_set_keepalive_device(ep);
```

Comment:
@dmitrygx why do we reset ep->keepalive.device here? If current device is invalid or comparison with cached keepalive.device doesn't match, we should throw an error, right?

Reply:
the idea is to initialize cached CUDA device in `uct_ep_check()` if it is invalid after `uct_ep_create()`.
and if we can cuCtxGetDevice is now OK, CUDA-IPC caches the value for further checking with a current one.

we need this, because a user may don't want to use CUDA memory at all, but we will have CUDA-IPC lane on which `uct_ep_check()` will be called (but CUDA device won't be initialized in this case)

Reply:
Makes sense is cudaipc ep creation can occur without a device context being present